clients/sellingpartner-api-aa-csharp/src/Amazon.SellingPartnerAPIAA/resources/swagger-codegen/templates/ApiClient.mustache

```diff
@@ -98,11 +98,33 @@ namespace {{packageName}}.Client
         public RestClient RestClient { get; set; }
 
         // Creates and sets up a RestRequest prior to a call.
+        // Wrapper on PrepareRequestHelper that proactively replaces pathParams to address the following issue:
+        // https://github.com/amzn/selling-partner-api-models/issues/42        
         private RestRequest PrepareRequest(
             String path, {{^netStandard}}RestSharp.{{/netStandard}}Method method, List<KeyValuePair<String, String>> queryParams, Object postBody,
             Dictionary<String, String> headerParams, Dictionary<String, String> formParams,
             Dictionary<String, FileParameter> fileParams, Dictionary<String, String> pathParams,
             String contentType)
+        {            
+            if (pathParams.Any())
+            {
+                var tempRequest = PrepareRequestHelper(path, method, queryParams, postBody, headerParams, formParams, fileParams, pathParams, contentType);
+
+                var pathWithVariablesReplaced = RestClient.BuildUri(tempRequest).LocalPath;
```

Comment:
You could put this direcly in `PrepareRequestHelper` instead of the current 
```cs
foreach(var param in pathParams)
     request.AddParameter(param.Key, param.Value, ParameterType.UrlSegment);
```